vncviewer/DesktopWindow.cxx

```diff
@@ -469,8 +473,14 @@ void DesktopWindow::fullscreen_on()
   fullscreen();
 }
 
+void DesktopWindow::toggle_grab_keys() {
+  this->grab_keys = !(this->grab_keys);
```

Comment:
This should be updated when the state changes, even if calling `grabKeyboard()` directly.